libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -818,6 +819,19 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 							if (suffix == "length")
 								stackDiff--;
 						}
+						else if (
+							auto const* functionType = dynamic_cast<FunctionType const*>(variable->type());
+							functionType && functionType->kind() == FunctionType::Kind::External
+						)
```

Comment:
This does not account for function variables stored in storage.

Reply:
That is intended. The ticket specification was "If x is a Solidity local variable of function external type [..]"